### PR TITLE
Auth pages

### LINKED ISSUE #3: Abstracted Server Validation
- [x] Abstract out the server validation function present on various pages
- [x] Replace with the stable version of Auth where needed

### PR DESCRIPTION
**Pull request by ispapa**
Closing #3 . Abstracts page validation.

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -78,9 +53,10 @@ const Home: NextPage<any> = () => {
...
         <RiFocus3Line />
       </button>
+      {/* This is where the Mapbox puts its stuff */}
       <div id="map" className="h-screen"></div>
     </>
   );
 };
 
-export default Home;
+export default () => ProtectedPage(Home);
```
Comment (dillydally414) [inline src/pages/index.tsx:62]: Can we add a comment here explaining why this arrow notation is necessary?

File: src/utils/auth.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { NextPage } from "next";
+import { useSession, getSession } from "next-auth/react";
+import { useRouter } from "next/router";
+import Spinner from "../components/Spinner";
+
+
+/**
```
Comment (dillydally414) [inline src/utils/auth.tsx:7]: Actually I take back the other comment, maybe we should add something in the JSDoc here for reference in future pages (so we know the arrow notation is necessary)? At least one or the other should be added though.

File: src/pages/onboard.tsx [hunk 3]
```diff
@@ -1,15 +1,13 @@
 import { Combobox, Transition } from "@headlessui/react";
```
Comment (dillydally414) [inline src/pages/onboard.tsx:1]: We should protect this page

File: src/pages/settings.tsx [hunk 4]
```diff
@@ -1,14 +1,13 @@
 import { Combobox, Transition } from "@headlessui/react";
```
Comment (dillydally414) [inline src/pages/settings.tsx:1]: We should also protect this page

**Review by dillydally414 — CHANGES_REQUESTED** [review]
The meat of this PR is good, just some small fixes requested :)

**Review by lukejianu — APPROVED** [review]
Looks good to me.

**Review by dillydally414 — APPROVED** [review]

**Review by devashish9 — APPROVED** [review]
Everything looks good, I'd be down to go through the exact changes that were needed to make the abstract validation work. Isabelle, you can merge this branch in during GM!

**Pull request merged**